daemon/algod/api/algod.oas2.json

```diff
@@ -934,6 +934,92 @@
         }
       }
     },
+    "/v2/participation/generate/{address}": {
+      "post": {
+        "tags": [
+          "private",
+          "participating"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "schemes": [
+          "http"
+        ],
+        "summary": "Generate and install participation keys to the node.",
+        "operationId": "GenerateParticipationKeys",
+        "parameters": [
+          {
+            "type": "string",
+            "description": "An account public key",
+            "name": "address",
+            "in": "path",
+            "required": true
+          },
+          {
+            "type": "integer",
+            "description": "Key dilution for two-level participation keys (defaults to sqrt of validity window).",
+            "name": "dilution",
+            "in": "query"
+          },
+          {
+            "type": "integer",
+            "description": "First round for participation key.",
+            "name": "first",
+            "in": "query",
+            "required": true
+          },
+          {
+            "type": "integer",
+            "description": "Last round for participation key.",
+            "name": "last",
+            "in": "query",
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The current swagger spec",
+            "schema": {
+              "type": "string"
```

Comment:
Have you considered making this endpoint return the participation ID of the keys that will be generated? That way you'd only need to query the `GET /v2/participation/{id}` endpoint, and it'd be very simple to determine if the entry were there or not. It seems like participation IDs are hashes of the keys' properties, so it may be possible.

Otherwise what would you need to do to confirm completion? Check that `GET /v2/participation/{id}` has an entry with the parameters that you submitted here?

Reply:
Yes, but the participation ID is not available. The success object is currently an empty JSON object with the idea that maybe better status becomes available in the future.